lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,75 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
```

Comment:
Most of the complications is in the C code, so read the PR description for a rundown of what happens.